modules/runner/src/test/java/org/apache/ignite/internal/component/RestAddressReporterTest.java

```diff
@@ -39,25 +39,53 @@ class RestAddressReporterTest {
     private static final String REST_ADDRESS_FILENAME = "rest-address";
 
     @Test
-    @DisplayName("REST server network address is reported to file")
-    void networkAddressReported(@TempDir Path tmpDir) throws IOException {
+    @DisplayName("REST server network addresses is reported to file")
+    void httpAndHttpsAddressesReported(@TempDir Path tmpDir) throws IOException {
         // Given
         RestAddressReporter reporter = new RestAddressReporter(tmpDir);
 
         // When
-        reporter.writeReport(new NetworkAddress("localhost", 9999));
+        reporter.writeReport(new NetworkAddress("localhost", 9999), new NetworkAddress("localhost", 8443));
+
+        // Then there is a report
+        String restAddress = Files.readString(tmpDir.resolve(REST_ADDRESS_FILENAME));
+        assertThat(restAddress, equalTo("http://localhost:9999, https://localhost:8443"));
+    }
+
+    @Test
+    @DisplayName("REST server HTTP address is reported to file")
+    void httpAddressReported(@TempDir Path tmpDir) throws IOException {
+        // Given
+        RestAddressReporter reporter = new RestAddressReporter(tmpDir);
+
+        // When
+        reporter.writeReport(new NetworkAddress("localhost", 9999), null);
 
         // Then there is a report
         String restAddress = Files.readString(tmpDir.resolve(REST_ADDRESS_FILENAME));
         assertThat(restAddress, equalTo("http://localhost:9999"));
     }
 
+    @Test
+    @DisplayName("REST server HTTPS address is reported to file")
+    void httpsAddressReported(@TempDir Path tmpDir) throws IOException {
+        // Given
+        RestAddressReporter reporter = new RestAddressReporter(tmpDir);
+
+        // When
+        reporter.writeReport(null, new NetworkAddress("localhost", 8443));
```

Comment:
changing the reporting string might corrupt the message that is printed after node start. could you please test it manually if ./ignite3db start printing readable message in case SSL is enabled?

Reply:
Now it looks like 
1. in the case of enabled SSL
Node named defaultNode started successfully. REST addresses are [http://127.0.0.1:10300, https://127.0.0.1:10400]
2. in case of disabled SSL
Node named defaultNode started successfully. REST addresses are [http://127.0.0.1:10300]